### ISSUE TITLE
[A11y] [Date picker] Buttons to navigate to next/previous months are announcing twice

### ISSUE THREAD

**Issue by darrenaustin** [flutter/flutter#96643]
When navigating the Date Picker under VoiceOver on iOS, when you activate the next/previous month button it will announce the button twice: 

> "Next month February 2022. Next month March 2022 button."

instead of the preferred: 

> "Next month March 2022 button."

Internal bug reference: b/213421262

<details>
  <summary>Example code</summary>

```dart
import 'dart:async';
import 'package:flutter/material.dart';

void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      home: MyHomePage(),
    );
  }
}

class MyHomePage extends StatefulWidget {
  @override
  _MyHomePageState createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  DateTime currentDate = DateTime.now();

  Future<void> _selectDate(BuildContext context) async {
    final pickedDate = await showDatePicker(
      context: context,
      initialDate: currentDate,
      firstDate: DateTime(2015),
      lastDate: DateTime(2050),
    );
    if (pickedDate != null && pickedDate != currentDate) {
      setState(() {
        currentDate = pickedDate;
      });
    }
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: const Text("DatePicker"),
      ),
      body: Center(
        child: Column(
          mainAxisSize: MainAxisSize.min,
          children: [
            Text(currentDate.toString()),
            ElevatedButton(
              onPressed: () => _selectDate(context),
              child: const Text('Select date'),
            ),
          ],
        ),
      ),
    );
  }
}
```
</details>

**Comment by darrenaustin** [comment]
This one is a bit tricky, as technically it is correct. When the button is activated, it announces its label (which happens to be 'Next month Feb 2022'), and then when the new month is showing, the button's label changes, so it announces the new label ('Next month March 2022 button').

That said, not a great experience. Also somewhere in there the new month should be announced by itself, but it probably getting stepped on by the button's announcement. I spent some time looking into this and I see two options:

1. Force the new month announcement to happen immediately after button presses and then let the button announce its new label. In this case the announcements would be: 

   > 'Feb 2022. Next month March 2022.'

2. Make the semantic label for the buttons just be 'Next/Prev Month' with no date. As they wouldn't be changing they would only announce when they are selected and not activated. Coupled with the new month announcement, activating the button would only yield:

    > 'Feb 2022.'

I am leaning toward the second option, but am open to suggestions and feedback here.

**Comment by darrenaustin** [comment]
cc @chunhtai in case you have any suggestions here.

**Comment by chunhtai** [comment]
The only guide line I can find is https://docs.google.com/document/d/186L8ngR0S-bAB_cU7Ph_YCRhBWlAgUbDjSMUF0PRPyc/edit?resourcekey=0-gsWhhjMlwVau59sfrId-GQ#heading=h.abjqhr42ibrt
and it looks like we should remove it.

I would double check with native behavior to see if they also does the same thing. If the native does not announce the tooltip after activate, we should just remove the announcement.

**Comment by darrenaustin** [comment]
I tested the native Android date picker on a Pixel 3 with Android 12 and it still has an older design. However its buttons just has the label 'Next month', and 'Previous Month' with no date.

I just posted a PR that removes the date from the next/previous button's semantics. This seems to resolve the problem in this case as the button's semantics don't change on the transition to the next month, so they don't re-announce themselves. I also moved the announcement for the newly displayed month so that it will override the the announcement for the button itself with more useful information.
